knodle/trainer/trainer.py

```diff
@@ -242,10 +242,14 @@ def test(
             self, features_dataset: TensorDataset, labels: TensorDataset, loss_calculation: bool = False
     ) -> Tuple[Dict, Union[float, None]]:
 
-        feature_label_dataset = input_labels_to_tensordataset(features_dataset, labels.tensors[0].cpu().numpy())
-        feature_label_dataloader = self._make_dataloader(feature_label_dataset, shuffle=False)
+        gold_labels = labels.tensors[0].cpu().numpy()
 
-        predictions, gold_labels, dev_loss = self._prediction_loop(feature_label_dataloader, loss_calculation)
+        if hasattr(self.model, "predict"):       # when the pytorch model is wrapped as a sklearn model (e.g. cleanlab)
```

Comment:
agree with `if isinstance(self.model, skorch.NeuralNetClassifier)`, but would rather have this loop here since the rest of the test function works for skorch classifier.